Formula/libgrapheme.rb

```diff
@@ -0,0 +1,43 @@
+class Libgrapheme < Formula
+  desc "Unicode string library"
+  homepage "https://libs.suckless.org/libgrapheme/"
+  url "https://dl.suckless.org/libgrapheme/libgrapheme-2.0.1.tar.gz"
+  sha256 "3db593a20fcbedcf4aa01c2ff973c430c40e19b961422ded95efe74a216cda2a"
+  license "ISC"
+  head "git://git.suckless.org/libgrapheme/", branch: "master"
+
+  def install
+    args = if OS.mac?
+      [
+        "PREFIX=#{prefix}",
+        "SOFLAGS=-dynamiclib -install_name 'libgrapheme.$(VERSION_MAJOR).dylib' " \
+        "-current_version '$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH)' " \
+        "-compatibility_version '$(VERSION_MAJOR).$(VERSION_MINOR).0'",
+        "SONAME=libgrapheme.$(VERSION_MAJOR).dylib",
+        "SOSYMLINK=false",
+        "LDCONFIG=",
+      ]
+    else
+      [
+        "PREFIX=#{prefix}",
+        "LDCONFIG=",
+      ]
+    end
```

Comment:
This just needs to be patched upstream to work correctly.

Reply:
Good morning, thanks for reviewing my submission!

I'm the upstream-developer. The program comes with a config.mk where this is given as such, and as you can see it's version-agnostic and won't change. It's simply setting the make-environment up, not a hack or anything.

It's not my fault that the macOS-linker needs this magic spell to work correctly with semantic library versioning (many programs do this wrong). The four variables SOFLAGS, SONAME, SOSYMLINK and LDCONFIG can perfectly encompass all major operating systems (any Linux, *BSD, macOS) and my goal is to give optimal flexibility. After all, even though it may look like a lot, the passed SOFLAGS are nothing complicated.

Those variables are obviously also not changing in future libgrapheme versions, so there won't be a maintenance burden for you. I promise that I will keep an eye on this brew-formula and send PRs for version bumps myself. As you can see, all tests are passing. :)

This library has already been packaged in other operating systems (OpenBSD, Alpine, Gentoo, Debian is underway) and this is the established way to do this with simple Makefile-based builds.

Reply:
Seeing how this is the only program I see that needs this list of variables, it definitely looks like a hack to me. Why not maintain build system specific config in the build system, something like:
```
	ifeq ($(UNAME_S),Darwin)
		SOFLAGS += <code here>
	endif
```

Reply:
Of course I would have never proposed such a solution if it was a hack, because I know that this really sucks for maintainers like you and wastes time. I looked before if this is also used in other formulae and was confirmed in this regard:

Please refer to the formulae [lua@5.1.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/lua@5.1.rb#L172), [lua@5.3.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/lua@5.3.rb#L168),  [jack.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/jack.rb#L44), [e2fsprogs.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/e2fsprogs.rb#L82) and [libgsm.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/libgsm.rb#L30), which also explicitly set compatibility_version, current_version and install_name. And I'm certain there are other formulae where this _should_ be set as the lack of such settings leads to silent linking problem on upgrades.

These three flags are not "necessary" to run libgrapheme and it will work fine without them. These flags are merely hints for the macOS-linker to respect semantic versioning (e.g. runtime-linking a dynamically linked binary compiled against 2.0 against all higher minor versions 2.1, 2.2, etc.).

Thus, I would like to kindly and respectfully ask you to reconsider.